File: algorithms/interview-preparation-kit-2/php/code-response/spec/FunnyString/FunnyString.spec.php
<?php

use FunnyString\FunnyString;;

describe('Funny String', function () {
  it('the string abc is funny', function () {
    expect(FunnyString::check('abc'))->toEqual('Funny');
  });

  it('the string abd is not funny', function () {
    expect(FunnyString::check('abd'))->toEqual('Not Funny');
  });

  it('the string acxz is funny', function () {
    expect(FunnyString::check('acxz'))->toEqual('Funny');
  });

  it('the string bcxz is not funny', function () {
    expect(FunnyString::check('bcxz'))->toEqual('Not Funny');
  });
});

File: algorithms/interview-preparation-kit-2/ruby/code-response/spec/FunnyString/funny_string.spec.rb
require_relative "../../src/FunnyString/funny_string"

describe "Funny String" do
  context "Is the string abc funny?" do
    it "The string abc is funny!" do
      expect(FunnyString.check('abc')).to eq('Funny')
    end
  end

  context "Is the string abd funny?" do
    it "The string abd is funny!" do
      expect(FunnyString.check('abd')).to eq('Not Funny')
    end
  end

  context "Is the string acxz funny?" do
    it "The string acxz is funny!" do
      expect(FunnyString.check('acxz')).to eq('Funny')
    end
  end

  context "Is the string bcxz funny?" do
    it "The string bcxz is funny!" do
      expect(FunnyString.check('bcxz')).to eq('Not Funny')
    end
  end

end